MWPW-134131 - Fixing issue when there are more than one rule with same number of grouping

sabyamon opened the pull request:
<!-- Before submitting, please review all open PRs. -->

Fixing the issue of the first rule getting picked when there was more than one rule with the same number of grouping.

Resolves: https://jira.corp.adobe.com/browse/MWPW-134131

**Test URLs:**
- Before: https://uar-integration--milo--adobecom.hlx.page/drafts/sabya/quiz-issue/?martech=off&hideGeorouting=on
- After: https://uar-same-number-group--milo--sabyamon.hlx.page/drafts/sabya/quiz-issue/?martech=off&hideGeorouting=on

chrischrischris commented on libs/blocks/quiz/utils.js:
```diff
@@ -317,24 +317,28 @@ export const findMatchForSelections = (results, selections) => {
-  const compoundResults = results.find((destination) => {
-    if (destination.result.indexOf('(') !== -1 && destination.result.split('&').length === userSelectionLen) {
-      return destination;
+  const probableMatches = [];
+  results.forEach((rule) => {
+    if (rule.result.indexOf('(') !== -1 && rule.result.split('&').length === userSelectionLen) {
+      probableMatches.push(rule);
     }
   });
```
nit
```suggestion
const probableMatches = results.reduce((match, rule) => {
  if (rule.result.includes('(') && rule.result.split('&').length === userSelectionLen) {
    match.push(rule);
  }
  return match;
}, []);
```

sabyamon replied:
Looks much better. Thanks for the feedback. Updated.

chrischrischris commented on libs/blocks/quiz/utils.js:
```diff
@@ -317,24 +317,28 @@ export const findMatchForSelections = (results, selections) => {
-      recommendations.push(compoundResults);
-      return recommendations;
+  probableMatches.forEach((rule) => {
+    const productList = rule !== undefined ? rule.result.split('&') : [];
```
nit: if `rule` is never `''` then you can just do `rule ? rule.result.split...`

sabyamon replied:
Thanks Chris! updated.

mokimo reviewed (approved):
Would be nice if you can fix some of the linting errors in this file (or skip the linter for the specific line if not otherwise possible)

overmyheadandbody reviewed (approved):
`console.log` call from L52 should probably be switched to a LANA log, same as any other calls to `console`

narcis-radu reviewed (approved):
Chris and Rareș have already left some suggestions, it would be great to implement them before merging. Thank you!

sabyamon commented:
> Would be nice if you can fix some of the linting errors in this file (or skip the linter for the specific line if not otherwise possible)

Thanks @mokimo for the review! Updated and made the linter happy for this file.

sabyamon commented:
> Chris and Rareș have already left some suggestions, it would be great to implement them before merging. Thank you!

Addressed all the review comments. It's immensely helping us before going to the `main` branch. Thanks all!
